tests/unity/example_test/src/example_test.c

```diff
@@ -45,7 +45,9 @@ void test_uut_init_with_param_check(void)
  */
```

Comment:
The above text and the `main` function can now be removed from all unit tests. 

We also no longer need the `unity_main()` and therefore the `main_name` entry can be removed from https://github.com/balaji-nordic/sdk-nrf/blob/e15119a2ad28f0755448477286f9ba8fa77a7ff6/tests/unity/unity_cfg.yaml.template#L37

Reply:
> We also no longer need the unity_main() 

Are you sure?

Zephyr main() functions currently must return 0. All other return values are reserved.

Reply:
I see. When a unit test fails, unity's `main()` will return the number of unit tests that failed. That would be a violation then. Would it be problem considering that we only run unit tests on `native_posix` board?

Reply:
> Would it be problem considering that we only run unit tests on native_posix board?

Would what be a problem?

It seems OK to continue to discard unity's main() return value and return 0 in our wrapper as required. Is there an issue I'm missing?

Reply:
> It seems OK to continue to discard unity's main() return value and return 0 in our wrapper as required. Is there an issue I'm missing?

That is fine then. We have always been ignoring the error from `unity_main()` main anyway. My only concern was that if there are any unit tests that fail, the exit code from the zephyr application should be non-zero. And it seems this code handles that https://github.com/nrfconnect/sdk-nrf/blob/b05fe3e9d0024d5677fd520de0f0f4fb4da4f9db/tests/unity/src/generic_teardown.c#L30 . 

> The above text and the main function can now be removed from all unit tests (atleast those in sdk-nrf).

This statement still holds though.

Reply:
>> The above text and the main function can now be removed from all unit tests (atleast those in sdk-nrf).
>
> This statement still holds though.

I aligned all the comments in our sdk-nrf tests that declare this extern.